src/merge.ts

```diff
@@ -0,0 +1,102 @@
+import { Packet, Receiver } from "./index";
+
+interface MergeProps {
+    universes?: number[];
+    port?: number;
+    iface?: string;
+    reuseAddr?: boolean;
+    timeout?: number;
+}
+
+export class ReceiverMerge extends Receiver {
+    constructor({
+        universes = [1],
+        port = 5568,
+        iface = undefined,
+        reuseAddr = false,
+        timeout = 5000,
+    }: MergeProps) {
+        super({
+            universes,
+            port,
+            iface,
+            reuseAddr
+        });
+        this.timeout = timeout;
+        super.on("packet", this.mergePacket);
+    }
+    readonly timeout: number;
+    protected senders = new Map<string, SendersData>();
+    protected lastData = new sACNData();
+    mergePacket(packet: Packet) {
+        // used to identify each source (cid & universe)
+        let pid: string = packet.cid.toString() + "#" + packet.universe.toString();
+        this.senders.set(
+            pid,
+            new SendersData(
+                packet.cid.toString(),
+                new sACNData(packet.payload),
+                packet.priority,
+                packet.sequence
+            ));
+        setTimeout(() => {
+            if (this.senders.get(pid)?.seq == packet.sequence) this.senders.delete(pid);
+        }, this.timeout);
+
+        // detect which source has the highest per-universe priority
+        let maximumPrio = 0;
+        for (let [_, data] of this.senders) {
+            if (data.prio > maximumPrio) {
+                maximumPrio = data.prio;
+            }
+        }
+
+        // HTP
+        let mergedData = new sACNData();
+        for (let [_, data] of this.senders) {
+            if (data.prio == maximumPrio) {
+                let i = 0;
+                while (i < 512) {
+                    let newValue = data.data.data[i];
+                    if (mergedData.data[i] < newValue) mergedData.data[i] = newValue;
+                    i++;
+                }
+            }
+        }
+
+        // console.log(mergedData);
+        // only changes
+        let i = 0;
+        while (i < 512) {
+            if (this.lastData.data[i] != mergedData.data[i]) {
+                super.emit("changed", {
```

Comment:
The other changes @k-yle souggested are quite usefull, but I am not a friend of this change.

Most users (99.9%) would do something like this:

````js
rec.on("changed", changes => {
  for(let change of changes){
    // do exactly the same as before
  }
})
````

In the few use cases, where caching is needed, you should use your own caching algorithm, to cache data from different universes and sources, too. This would be more efficient.

The only reason to implement it like this could be the Node.js native EventEmitter's performance; but the word *performance* doesn't appear in the official documentation.
If you think, that this point is important, please change it.

Reply:
I can understand your thought, that too many emitted events could affect the performance negatively.
A simple solution could be to move the `EventEmitter.emit()` call into a `process.nextTick`.
This way the event would be emitted asynchronously.

To improve external caching, we could emit a second event which indicates, that the full universe has been merged.

What do you think about this?